src/com/amazon/corretto/crypto/provider/RsaCipher.java

```diff
@@ -211,25 +234,119 @@ protected AlgorithmParameters engineGetParameters() {
+                if (nativeKey_ != null) {
+                    nativeKey_.releaseEphemeral();
+                    nativeKey_ = null;
+                }
+
+                key_ = (RSAKey) key;
+                keySizeBytes_ = (key_.getModulus().bitLength() + 7) / 8;
+                buffer_ = new AccessibleByteArrayOutputStream();
+                nativeKey_ = provider_.translateKey(key, EvpKeyType.RSA);
+            }
+
+            if (params instanceof OAEPParameterSpec) {
+                final OAEPParameterSpec oaepParams = (OAEPParameterSpec) params;
+                try {
+                    // Cache MD struct ptrs, validate digest names, update params + padding len
+                    getMdPtr(oaepParams.getDigestAlgorithm());
+                    getMdPtr(((MGF1ParameterSpec) oaepParams.getMGFParameters()).getDigestAlgorithm());
+                    paddingSize_ = calculateOaepPaddingLen(oaepParams.getDigestAlgorithm());
+                    oaepParams_ = oaepParams;
+                } catch (Exception e) {
+                    throw new InvalidAlgorithmParameterException();
+                }
+            }
+        }
+    }
+
+    private static String jceNameToAwsLcName(String jceName) {
+        if (jceName == null) {
+            return null;
+        }
+        // e.g. "SHA-512/256" => "SHA512-256"
+        return jceName.replace("-", "").replace("/", "-").toUpperCase();
+    }
+
+    private static long getMdPtr(String digestName) {
+        final String name = jceNameToAwsLcName(digestName);
+        return digestPtrByName.computeIfAbsent(name, n -> Utils.getEvpMdFromName(n));
+    }
+
+    private static int getMdLen(long mdPtr) {
+        return digestLengthByPtr.computeIfAbsent(mdPtr, p -> Utils.getDigestLength(p));
+    }
+
+    // NOTE: while RFC-2437 stipulates[1] that OAEP padding has max length of 2*digestSize+1,
+    //       both standard JCE[2] and AWS-LC[3][4] reserve an extra byte of padding space to
+    //       ensure that the size of the padded message does not excede that of the modulus,
+    //       hence the +2 in the calculation below.
+    //
+    // [1]: https://datatracker.ietf.org/doc/html/rfc2437#section-9.1.1.1
+    // [2]: https://github.com/corretto/corretto-8/blob/develop/src/jdk/src/share/classes/sun/security/rsa/RSAPadding.java#L191
+    // [3]: https://github.com/awslabs/aws-lc/blob/main/crypto/fipsmodule/rsa/padding.c#L349
+    // [4]: https://github.com/awslabs/aws-lc/blob/main/crypto/fipsmodule/rsa/padding.c#L420-L427
+    private static int calculateOaepPaddingLen(String mdName) {
+        return 2 * getMdLen(getMdPtr(mdName)) + 2;
+    }
+
+    @Override
+    protected void engineInit(final int opmode, final Key key, final AlgorithmParameters params,
+            final SecureRandom random)
+                    throws InvalidKeyException, InvalidAlgorithmParameterException {
+        try {
+            engineInit(opmode, key, params.getParameterSpec(OAEPParameterSpec.class), random);
```

Comment:
Is it possible for `params` to be null?

Reply:
i don't think so, as the [prior logic](https://github.com/corretto/amazon-corretto-crypto-provider/blob/develop/src/com/amazon/corretto/crypto/provider/RsaCipher.java#L310) also assumed this was non-null. but it wouldn't hurt to do a null check here. i'll add it.